toml.md

```diff
@@ -104,27 +104,44 @@ Keys
 
 A key may be either bare, quoted, or dotted.
 
-**Bare keys** may only contain ASCII letters, ASCII digits, underscores, and
-dashes (`A-Za-z0-9_-`). Note that bare keys are allowed to be composed of only
-ASCII digits, e.g. `1234`, but are always interpreted as strings.
+**Bare keys** may contain any letter-like or number-like Unicode character from
+any Unicode script, as well as ASCII digits, dashes and underscores. 
+Punctuation, spaces, arrows, box drawing and private use characters are not
+allowed.  Note that bare keys are allowed to be composed of only ASCII digits,
+e.g. `1234`, but are always interpreted as strings.
+
+* Of the ASCII characters, only `A-Z`, `a-z`, `0-9`, `_` and `-` are allowed.
+* Of the rest of the ISO-8859-1 characters, allowed are U+B2, U+B3, U+B9, 
+  U+BC to U+BE, U+C0 to U+D6, U+D8 to U+F6 and U+F8 to U+FF
+* All of U+0100-1FFF are allowed, except `;` (U+037E).
+* Characters U+200C, U+200D, U+203F, U+2040, U+2070 to U+218F, U+2460 to 
+  U+24FF, and U+2C00 to U+2FEF are allowed.
+* All characters from U+3001 and higher are allowed, except surrogates (U+D800 to U+DFFF),
+  Private Use (U+E000 to U+F8FF, U+F0000 to U+10FFFF), non-characters (U+FFFE 
+  and U+FFFF) and process-internal use (U+FDD0 to U+FDEF).
```

Comment:
I don't think we should be listing this content here. I think having plain digestable language here is much more important than having the spec mirror the details that the ABNF would then unambigously declare.

I notice that this was added since @ChristianSi said:

> As I see it, the written spec should be self-contained and one should not have to read the ABNF to find out what it actually means. 

And... I disagree with that. The whole point of the spec is to be human-readable and including such a dense description with unicode codepoints right at the start is... uhm... suboptimal.

Reply:
@pradyunsg i can see your point. I’m not necessarily in favour of either approach. I could certainly replace the whole bulleted list with something that says “all natural letters and letterlike characters, all digits in all languages”.

Maybe you have something specific in mind?

Reply:
Would it be sufficient to draw explicit attention to the ABNF here, as an inset or similar? Something like:

> ℹ&#xFE0F; _The exact ranges of allowed characters can be found in the [ABNF grammar file](https://github.com/toml-lang/toml/blob/main/toml.abnf)._

As it stands the ABNF is only mentioned as an afterthought wayyyyyy at the bottom of the spec document. If it is to be used as an authoritative complement to the spec, we probably should mention it earlier/more prominently.